frontend/lib/src/components/widgets/AudioInput/uploadFiles.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import zip from "lodash/zip"
+
+import { FileUploadClient } from "@streamlit/lib/src/FileUploadClient"
+import {
+  WidgetInfo,
+  WidgetStateManager,
+} from "@streamlit/lib/src/WidgetStateManager"
+import {
+  FileUploaderState as FileUploaderStateProto,
+  IFileURLs,
+  UploadedFileInfo as UploadedFileInfoProto,
+} from "@streamlit/lib/src/proto"
+import { ensureError } from "@streamlit/lib/src/util/ErrorHandling"
+
+type SuccessfulUpload = {
+  fileUrl: IFileURLs
+  file: File
+}
+
+type FailedUpload = {
+  file: File
+  error: Error
+}
+
+export const uploadFiles = async ({
+  files,
+  uploadClient,
+  widgetMgr,
+  widgetInfo,
+}: {
+  files: File[]
+  uploadClient: FileUploadClient
+  widgetMgr: WidgetStateManager
+  widgetInfo: WidgetInfo
+}): Promise<{
+  successfulUploads: SuccessfulUpload[]
+  failedUploads: FailedUpload[]
+}> => {
+  let fileUrls: IFileURLs[] = []
+
+  try {
+    fileUrls = await uploadClient.fetchFileURLs(files)
+  } catch (e) {
+    console.error("Error fetching file URLs")
```

Comment:
**question:** Do we typically keep these kinds of logs in?

Reply:
Nah that was just missed